src/AWS.Deploy.DockerImageUploader/App.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using AWS.Deploy.Common.IO;
+using AWS.Deploy.Common;
+using Microsoft.Extensions.DependencyInjection;
+using System.Reflection;
+using System.Collections.Generic;
+using System;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace AWS.Deploy.DockerImageUploader
+{
+    public class App
+    {
+        private readonly IFileManager _fileManager;
+        private readonly IDirectoryManager _directoryManager;
+        private readonly IProjectDefinitionParser _projectDefinitionParser;
+        private readonly CLI.App _deployToolCli;
+
+        private readonly List<string> _testApps = new List<string> { "WebApiNET6", "ConsoleAppTask" };
+
+        public App(IServiceProvider serviceProvider)
+        {
+            _projectDefinitionParser = serviceProvider.GetRequiredService<IProjectDefinitionParser>();
+            _fileManager = serviceProvider.GetRequiredService<IFileManager>();
+            _directoryManager = serviceProvider.GetRequiredService<IDirectoryManager>();
+            _deployToolCli = serviceProvider.GetRequiredService<CLI.App>();
+        }
+
+        public async Task Run(string[] args)
+        {
+            foreach (var testApp in _testApps)
+            {
+                var projectPath = ResolvePath(testApp);
+                await CreateImageAndPushToECR(projectPath);
+            }
+        }
+
+        private async Task CreateImageAndPushToECR(string projectPath)
+        {
+            var projectDefinition = await _projectDefinitionParser.Parse(projectPath);
+
+            var dockerEngine = new DockerEngine.DockerEngine(projectDefinition, _fileManager, _directoryManager);
+            dockerEngine.GenerateDockerFile();
+
+            var configFilePath = Path.Combine(projectPath, "DockerImageUploaderConfigFile.json");
+            var deployArgs = new[] { "deploy", "--project-path", projectPath, "--diagnostics", "--apply", configFilePath, "--silent" };
```

Comment:
Nevermind, I forgot that we don't offer to generate the Dockerfile for the ECR recipe, only for Fargate and App Runner. So now see that the console app is necessary so we can invoke `GenerateDockerFile`.

Reply:
Yeah, the _"PushToECR"_ recipe does not generate a Dockerfile. But even if it did, directly using the CLI would not serve our use case.

The GitHub action that drives his process is configured to run on push to `main`. If we use the CLI directly, it will use the Dockerfile template from an already released version of the tool, ignoring any updates that the new commits on `main` contain.